website/blog/2022/04/17/md.md

```diff
@@ -0,0 +1,313 @@
+
+[![prometheus-plugin-grafana-dashboard-screenshot](https://res.cloudinary.com/practicaldev/image/fetch/s--4-6fmBbW--/c_limit%2Cf_auto%2Cfl_progressive%2Cq_auto%2Cw_880/https://dev-to-uploads.s3.amazonaws.com/uploads/articles/mdpwdm84lqmbd9cyytt0.png)](https://res.cloudinary.com/practicaldev/image/fetch/s--4-6fmBbW--/c_limit%2Cf_auto%2Cfl_progressive%2Cq_auto%2Cw_880/https://dev-to-uploads.s3.amazonaws.com/uploads/articles/mdpwdm84lqmbd9cyytt0.png)
+
+You can also check two other plugins for metrics:
+
+* [Skywalking-plugin](https://apisix.apache.org/docs/apisix/plugins/skywalking)
+
+* [Opentelemetry-plugin](https://apisix.apache.org/docs/apisix/plugins/opentelemetry)
+
+## Tracing
+
+The third is **tracing** or distributed tracing allows you to understand the life of a request as it traverses your service network and allows you to answer questions like what service has this request touched and how much latency was introduced. Traces enable you to further explore which logs to look at for a particular session or related set of API calls.
+
+[Zipkin](https://zipkin.io/) an open-source distributed tracing system. [APISIX plugin](https://apisix.apache.org/docs/apisix/plugins/zipkin) is supported to collect tracing and report to Zipkin Collector based on [Zipkin API specification](https://zipkin.io/pages/instrumenting.html).
+
+Here’s an example to enable the _zipkin plugin_ on the specified route:
+
+```
+    curl http://127.0.0.1:9080/apisix/admin/routes/1  -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+
+{
+
+    "methods": ["GET"],
+
+    "uri": "/get",
+
+    "plugins": {
+
+        "zipkin": {
+
+            "endpoint": "http://127.0.0.1:9411/api/v2/spans",
+
+            "sample_ratio": 1
+
+        }
+
+    },
+
+    "upstream_id": "1"
+
+}'
+
+```
+
+We can test our example by simply running the following curl command:
+
+`curl -i http://127.0.0.1:9080/get`
+
+As you can see, there are some additional trace identifiers (like traceId, spanId, parentId) were appended to the headers:
+
+```
+    "X-B3-Parentspanid": "61bd3f4046a800e7",
+
+    "X-B3-Sampled": "1",
+
+    "X-B3-Spanid": "855cd5465957f414",
+
+    "X-B3-Traceid": "e18985df47dab632d62083fd96626692",
+
+```
+
+Then you can use a browser to access `http://127.0.0.1:9411/zipkin`, see traces on the Web UI of Zipkin.
+
+> Note that you need to run the Zipkin instance in order to install Zipkin Web UI. For example, by using docker you can simply run it:
+> `docker run -d -p 9411:9411 openzipkin/zipkin`
+
+[![Zipkin plugin output 1](https://res.cloudinary.com/practicaldev/image/fetch/s---BmZMyXZ--/c_limit%2Cf_auto%2Cfl_progressive%2Cq_auto%2Cw_880/https://dev-to-uploads.s3.amazonaws.com/uploads/articles/av8rdcwwybyfcibbo7xh.png)](https://res.cloudinary.com/practicaldev/image/fetch/s---BmZMyXZ--/c_limit%2Cf_auto%2Cfl_progressive%2Cq_auto%2Cw_880/https://dev-to-uploads.s3.amazonaws.com/uploads/articles/av8rdcwwybyfcibbo7xh.png)
```

Comment:
@guoqqqi Please help to upload images to CDN. Or @hanzhenfang, you could open an issue (don't submit), upload pictures, and get the image URL.

Reply:
I have uploaded those pics and left link in comments